### PR TITLE
AI Alliance RAG Demo

### PR DESCRIPTION
**Pull request by touma-I**
## Why are these changes needed?

Implement notebook from the from AI Alliance for RAG demo.

### CODE REVIEW

File: examples/notebooks/rag/Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+REPOROOT=../../..
+# Use make help, to see the available rules
+
+include $(REPOROOT)/.make.defaults
+
+clean:: .defaults.clean
+
+venv:: .defaults.ray-lib-src-venv
+	@# Help: Install requirements from all transforms into a venv 
+	. ./venv/bin/activate && bash ./prepare_env.sh
```
Comment (touma-I) [inline examples/notebooks/rag/Makefile]:
```suggestion
	. ./venv/bin/activate && pip install data-prep-toolkit-transforms-ray==0.2.1.dev1
```

File: examples/notebooks/rag/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+## Torch and enbeddings
+torch
+sentence-transformers
+
+## --- Parquet
+pandas
+pyarrow
+
+## DPK
```
Comment (touma-I) [inline examples/notebooks/rag/requirements.txt]: DPK dependencies should be already installed when we do  `pip install data-prep-toolkit-transforms-ray==0.2.1.dev1`

**Review by touma-I — COMMENTED** [review]
@sujee Let's have a quick call. We may be able to make things a bit simpler when it comes to setting up the environment and installing the DPK library and transform. Please give me your availability for tomorrow Thursday for a 30 minute Teams session.
cc: @shahrokhDaijavad

**Comment by shahrokhDaijavad** [comment]
Will be happy to join the call tomorrow, @sujee and @touma-I .

File: examples/notebooks/rag/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
+# RAG with Data Prep Kit
+
+This folder has examples of RAG applications with data prep kit (DPK).
+
+## Step-0: Getting Started
+
+```bash
+## Clone this repo
+git  clone   https://github.com/sujee/data-prep-kit
```
Comment (touma-I) [inline examples/notebooks/rag/README.md]: Once this is merged with the main branch, do we want to maintain this fork ? If we do not maintain the fork, the URL will need to change to https://github.ibm.com/IBM/data-prep-kit

Comment (shahrokhDaijavad) [reply examples/notebooks/rag/README.md]: @touma-I Typo in the URL above: https://github.com/IBM/data-prep-kit

@sujee, I tried running your branch and got an error. I used Option B in examples/notebooks/rag/setup-python-dev-env.md to setup the Python dev environment and then started rag_1A_dpk_process_ray.ipynb in the Jupyter notebook, but something about the DPK imports is not right and I get the error below:
image

Comment (sujee) [reply examples/notebooks/rag/README.md]: > Once this is merged with the main branch, do we want to maintain this fork ? If we do not maintain the fork, the URL will need to change to https://github.ibm.com/IBM/data-prep-kit

Good point.  will update to reflect the main repo!

Comment (sujee) [reply examples/notebooks/rag/README.md]: > @touma-I Typo in the URL above: https://github.com/IBM/data-prep-kit
> 
> @sujee, I tried running your branch and got an error. I used Option B in examples/notebooks/rag/setup-python-dev-env.md to setup the Python dev environment and then started rag_1A_dpk_process_ray.ipynb in the Jupyter notebook, but something about the DPK imports is not right and I get the error below: image

Let me check , in the mean time check these for me please...

- once you activate your venv
- from CLI: python version either v3.10 or 3.11    (`python -version`  and `which python`  and `which pip`)
- `! pip list` in notebook and see if we have the package is present

Comment (shahrokhDaijavad) [reply examples/notebooks/rag/README.md]: @sujee python version is 3.11. 

which python
/Users/shahrokhdaijavad/Documents/GitHub/data-prep-kit-rag-example1/data-prep-kit/examples/notebooks/rag/venv/bin/python
(venv) (base) shahrokhdaijavad@shahrokhs-mbp rag % which pip
/Users/shahrokhdaijavad/Documents/GitHub/data-prep-kit-rag-example1/data-prep-kit/examples/notebooks/rag/venv/bin/pip


`! pip list ` in the notebook shows the following:

data_prep_toolkit                       0.2.1.dev0
data_prep_toolkit_ray                   0.2.1.dev0
data_prep_toolkit_transforms            0.2.1.dev1
data_prep_toolkit_transforms_ray        0.2.1.dev1

Comment (shahrokhDaijavad) [reply examples/notebooks/rag/README.md]: @sujee I assume you don't see this error in your environment, so if you want to test a few other things in my environment to find the source of the problem, please ping me and we can have a quick session of sharing my screen and trying things.

Comment (shahrokhDaijavad) [reply examples/notebooks/rag/README.md]: @sujee Just did a test of going through all Notebooks (all the way to the end of Step 6 in README) and verified that everything worked. 
@touma-I I think this is ready to pull into dev.

**Review by shahrokhDaijavad — APPROVED** [review]
LGTM

**Review by shahrokhDaijavad — CHANGES_REQUESTED** [review]
Shahrokh needs to check the fact the pdf Walmart files are "open" and not proprietary. 

The two Walmart files used in the example are available publicly at: 

https://corporate.walmart.com/content/dam/corporate/documents/newsroom/2024/04/25/walmart-releases-2024-annual-report-and-proxy-statement/walmart-inc-2024-annual-report.pdf

and 

https://s201.q4cdn.com/262069030/files/doc_financials/2023/ar/Walmart-10K-Reports-Optimized.pdf

@sujee Can we get WGET these files to your Notebook on the fly and not have them as part of the repo?

**Comment by shahrokhDaijavad** [comment]
Thanks, @sujee!

**Review by shahrokhDaijavad — APPROVED** [review]

**Review by shahrokhDaijavad — APPROVED** [review]
The changes LGTM.

**Pull request merged**